nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -312,26 +327,33 @@ namespace, getRef().getName()),
   }
 
   public void renameTable(TableIdentifier from, TableIdentifier to) {
+    renameContent(from, to, Content.Type.ICEBERG_TABLE);
+  }
+
+  public void renameView(TableIdentifier from, TableIdentifier to) {
+    renameContent(from, to, Content.Type.ICEBERG_VIEW);
+  }
+
+  private void renameContent(TableIdentifier from, TableIdentifier to, Content.Type type) {
     getRef().checkMutable();
 
-    IcebergTable existingFromTable = table(from);
-    if (existingFromTable == null) {
-      throw new NoSuchTableException("Table does not exist: %s", from.name());
-    }
-    IcebergTable existingToTable = table(to);
-    if (existingToTable != null) {
-      throw new AlreadyExistsException("Table already exists: %s", to.name());
-    }
+    IcebergContent existingFromContent = asContent(from, IcebergContent.class);
+    validateFromContentForRename(from, type, existingFromContent);
+
+    IcebergContent existingToContent = asContent(to, IcebergContent.class);
+    validateToContentForRename(from, to, existingToContent);
 
+    String contentTypeString = type == Content.Type.ICEBERG_VIEW ? "view" : "table";
```

Comment:
nit: maybe rename to `contentType` instead of using the `String` suffix, since you're already using the same var name in `dropContent()`